### PR TITLE
topolog2: nocodec: set output_pin number

### PR DESCRIPTION
**Pull request by RanderWang**
The num_output_pins was set for pass-through path but missed in normal path. Without it Linux kernel driver wouldn't set second output pin format for dai copier which connects smart amp and gain module.

### CODE REVIEW

File: tools/topology/topology2/cavs-nocodec.conf [hunk 1]
```diff
@@ -375,6 +375,7 @@ IncludeByKey.PASSTHROUGH {
 						out_valid_bit_depth	32
 					}
 
+					num_output_pins 2
```
Comment (plbossart) [inline tools/topology/topology2/cavs-nocodec.conf:378]: we should also clean-up the audio format part @RanderWang, we need to have different structures for in and out. And I also don't get why the format should be the same on all pins?

**Comment by RanderWang** [comment]
> is there a link to the issue this PR fixes?

No. It is found in loadable smart amp validation. I also check the built-in smart amp. The format setting by set_large_config is also missed for this reason.

**Comment by RanderWang** [comment]
Clean up audio format in cavs-nocodec

File: tools/topology/topology2/cavs-nocodec.conf (hunk 1 above)
Comment (RanderWang) [reply tools/topology/topology2/cavs-nocodec.conf:392]: Add clean-up.  @aiChaoSONG  can you help to answer pin format question ?

**Review by kv2019i — APPROVED** [review]
Not sure it's good idea to couple a bugfix and cleanup of audio formatting in same PR, but given you already did the work, it seems good.

File: tools/topology/topology2/cavs-nocodec.conf [hunk 2]
```diff
@@ -307,45 +307,54 @@ IncludeByKey.PASSTHROUGH {
...
-							num_output_audio_formats 2
-							Object.Base.audio_format.1 {
+							Object.Base.input_audio_format.1 {
 								in_bit_depth		32
 								in_valid_bit_depth	32
-								out_bit_depth		32
-								out_valid_bit_depth	32
 							}
-							Object.Base.audio_format.2 {
+							Object.Base.input_audio_format.2 {
```
Comment (ranj063) [inline tools/topology/topology2/cavs-nocodec.conf:316]: @RanderWang can you please avoid the .1/.2 and use array instead?

**Comment by RanderWang** [comment]
> Not sure it's good idea to couple a bugfix and cleanup of audio formatting in same PR, but given you already did the work, it seems good.

Good idea, thanks!

File: tools/topology/topology2/cavs-nocodec.conf (hunk 2 above)
Comment (RanderWang) [reply tools/topology/topology2/cavs-nocodec.conf]: sure, thanks

**Comment by RanderWang** [comment]
use audio format array, thanks!

**Review by ranj063 — APPROVED** [review]

**Review by plbossart — COMMENTED** [review]
I'll let @ranj063 review, and still don't get why the formats should be the same on all pins.

File: tools/topology/topology2/cavs-nocodec.conf (hunk 1 above)
Comment (ranj063) [reply tools/topology/topology2/cavs-nocodec.conf:407]: @plbossart both the gain and the smart amp only support 32-bit in this case.

**Pull request merged**